manifests/common/plumgrid.pp

```diff
@@ -11,6 +14,16 @@
     owner   => root,
     group   => root,
     mode    => '0440',
-    content => 'nova ALL=(root) NOPASSWD: /opt/pg/bin/ifc_ctl_pp *\n',
+    content => "nova ALL=(root) NOPASSWD: /opt/pg/bin/ifc_ctl_pp *\n",
+  }
+
+  openstack::resources::firewall { 'Nova API': port => '8774', }
+  openstack::resources::firewall { 'Nova Metadata': port => '8775', }
+
+  class { '::nova::api':
```

Comment:
I would prefer to prioritize separation of concerns over bundling things together for convenience. If you need to manage the Nova API on the same node, that can be done in your site's manifests.